File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        int cnt = 0, n = nums.size();
        vector<int>occX;
        for(int i=0; i<n; i++) {
            if(nums[i] == x) occX.push_back(i);
        }
        vector<int>ans;
        for(int q:queries) {
            if(q > occX.size()) ans.push_back(-1);
            else ans.push_back(occX[q-1]);
        }
        return ans;
    }
};

File: 3159-find-occurrences-of-an-element-in-an-array/3159-find-occurrences-of-an-element-in-an-array.java
class Solution {
    public int[] occurrencesOfElement(int[] nums, int[] queries, int x) {
        List<Integer>list = new ArrayList<>();
        for(int i=0; i<nums.length; i++) {
            if(nums[i] == x) list.add(i);
        }
        int n = queries.length;
        int[] ans = new int[n];
        for(int i=0; i<n; i++) {
            if(queries[i] > list.size()) ans[i] = -1;
            else ans[i] = list.get(queries[i] - 1);
        }
        return ans;
    }
}